### PR TITLE
Add Command to Remove EventListener

### PR DESCRIPTION
**Pull request by khrm**
Add Command to Remove EventListener

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [x] Run the code checkers with `make check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/eventlistener/delete.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/tektoncd/cli/pkg/helper/options"
+	"github.com/tektoncd/cli/pkg/helper/validate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+func deleteCommand(p cli.Params) *cobra.Command {
+	opts := &options.DeleteOptions{Resource: "eventlistener", ForceDelete: false}
+	f := cliopts.NewPrintFlags("delete")
+	eg := `Delete a EventListener of name 'foo' in namespace 'bar'
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete.go]:
```suggestion
	eg := `Delete an EventListener of name 'foo' in namespace 'bar'
```

File: pkg/cmd/eventlistener/delete.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+or
+
+    tkn el rm foo -n bar
+`
+
+	c := &cobra.Command{
+		Use:          "delete",
+		Aliases:      []string{"rm"},
+		Short:        "Delete a eventlistener in a namespace",
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete.go]:
```suggestion
		Short:        "Delete an eventlistener in a namespace",
```

File: pkg/cmd/eventlistener/delete.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+func deleteEventListener(s *cli.Stream, p cli.Params, elName string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return fmt.Errorf("failed to create tekton client")
+	}
+
+	if err := cs.Triggers.TektonV1alpha1().EventListeners(p.Namespace()).Delete(elName, &metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("failed to delete eventlistener %q: %s", elName, err)
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete.go]:
```suggestion
		return fmt.Errorf("failed to delete EventListener %q: %s", elName, err)
```

File: pkg/cmd/eventlistener/delete_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			wantError:   false,
+			want:        "EventListener deleted: el-1\n",
+		},
+		{
+			name:        "Without force delete flag, reply no",
+			command:     []string{"rm", "el-1", "-n", "ns"},
+			input:       seeds[2],
+			inputStream: strings.NewReader("n"),
+			wantError:   true,
+			want:        "canceled deleting eventlistener \"el-1\"",
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete_test.go]:
```suggestion
			want:        "canceled deleting EventListener \"el-1\"",
```

Can we capitalize this to make consistent with rest of command?

File: pkg/cmd/eventlistener/delete_test.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+			wantError:   false,
+			want:        "Are you sure you want to delete eventlistener \"el-1\" (y/n): EventListener deleted: el-1\n",
+		},
+		{
+			name:        "Remove non existent resource",
+			command:     []string{"rm", "nonexistent", "-n", "ns"},
+			input:       seeds[2],
+			inputStream: strings.NewReader("y"),
+			wantError:   true,
+			want:        "failed to delete eventlistener \"nonexistent\": eventlisteners.tekton.dev \"nonexistent\" not found",
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete_test.go]:
```suggestion
			want:        "failed to delete EventListener \"nonexistent\": eventlisteners.tekton.dev \"nonexistent\" not found",
```

File: pkg/cmd/eventlistener/delete_test.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+			wantError:   true,
+			want:        "canceled deleting eventlistener \"el-1\"",
+		},
+		{
+			name:        "Without force delete flag, reply yes",
+			command:     []string{"rm", "el-1", "-n", "ns"},
+			input:       seeds[2],
+			inputStream: strings.NewReader("y"),
+			wantError:   false,
+			want:        "Are you sure you want to delete eventlistener \"el-1\" (y/n): EventListener deleted: el-1\n",
```
Comment (danielhelfand) [inline pkg/cmd/eventlistener/delete_test.go]:
```suggestion
			want:        "Are you sure you want to delete EventListener \"el-1\" (y/n): EventListener deleted: el-1\n",
```

Can we capitalize this to make consistent with rest of command?

Comment (khrm) [reply pkg/cmd/eventlistener/delete_test.go]: This is coming from options. If we change here, then we would need to change in other commands also.

File: pkg/cmd/eventlistener/delete_test.go (hunk 4 above)
Comment (khrm) [reply pkg/cmd/eventlistener/delete_test.go]: This is coming from options. If we change here, then we would need to change in other commands also.

File: pkg/cmd/eventlistener/delete.go (hunk 3 above)
Comment (khrm) [reply pkg/cmd/eventlistener/delete.go]: This makes it command inconsistent with commands for other resources.

Comment (danielhelfand) [reply pkg/cmd/eventlistener/delete.go]: Good call. We should update for other resources then, but we can do that in a separate pr.

**Pull request merged**